frontend/src/pages/EditDepartment/EditDepartment.tsx

```diff
@@ -0,0 +1,162 @@
+  const [wardType, setWardType] = React.useState<WardType>();
+
+  const { hospitalId, departmentId } = useParams();
+
+  React.useEffect(() => {
+    if (hospitalId && departmentId) {
+      (async () => {
+        const hospitalResponse = await getHospital(hospitalId);
+        const departmentResponse = await getDepartment(departmentId);
+        const wardTypesResponse = await getWardTypes();
+
+        setHospitalData(hospitalResponse);
+        setDepartmentData(departmentResponse);
+        setWardTypesList(wardTypesResponse);
+        setWardType(wardTypesResponse[0].name as WardType);
+      })();
+    }
+  }, []);
+
+  const handleFormSubmit = async () => {
+    if (wardName && wardName.length > 0 && departmentData.id && wardType) {
+      const response = await addWard({
+        name: wardName,
+        departmentId: departmentData.id,
+        beds: [],
+        wardType
+      });
+
+      const newWardData = [response, ...departmentData.wards];
+
+      setDepartmentData({
+        ...departmentData,
+        wards: newWardData
+      });
+    } else {
+      console.error("Missing POST-Body Data");
+    }
+  };
+
+  const handleDelete = async (wardId: string) => {
+    const response = await deleteWard(wardId);
+
+    if (response.status === 200) {
+      // Remove deleted Department from State-Data on successfull DELETE
+      const newWardData = departmentData.wards.filter((ward: Ward) => ward.id !== wardId);
+
+      setDepartmentData({
+        ...departmentData,
+        wards: newWardData
+      });
+    }
+  };
+
+  return (
+    <Container>
+      {hospitalData && departmentData && (
+        <>
+          <Row>
+            <Col>
+              <Breadcrumb>
+                <LinkContainer to={`/hospital/${hospitalId}/edit`}>
+                  <Breadcrumb.Item>{hospitalData.name}</Breadcrumb.Item>
+                </LinkContainer>
+                <Breadcrumb.Item active>{departmentData.name}</Breadcrumb.Item>
+              </Breadcrumb>
+            </Col>
+          </Row>
+          <Row className="mb-4">
+            <Col>
+              <h3>Stationen editieren</h3>
+            </Col>
+          </Row>
+          <Row>
+            <Col>
+              {departmentData.wards.length > 0 ? (
+                departmentData.wards.map((ward: Ward) => (
+                  <CardData
+                    key={ward.id}
+                    title={ward.name}
+                    subTitle={`Beds: ${ward.beds ? ward.beds.length : 0}`}
+                    mainId={departmentData.id}
+                    subId={ward.id}
+                    editLink={`/hospital/${hospitalData.id}/${departmentData.id}/${ward.id}/edit`}
+                    deleteFunc={() => handleDelete(ward.id)}
+                  />
+                ))
+              ) : (
+                <p className="text-muted">Keine Stationen angelegt</p>
+              )}
+            </Col>
+            <Col>
+              <Card>
```

Comment:
Could we wrap this into a new React component? We could actually reuse the functionality for `add_department` then 😉